app/models/miq_filter.rb

```diff
@@ -101,11 +101,14 @@ def self.apply_belongsto_filters(inputs, bfilters)
     return []     if inputs.nil?
     return inputs if bfilters.empty?
 
+    vcmeta_index = bfilters.index_with { |tag| belongsto2object_list(tag) }
```

Comment:
I expected you to pass in a cache here so the parent objects were not looked up again and again. I was even thinking a relationship hash but an object pair map (Ems:1) may make more sense for little baby's feet

are you thinking this is the right object to cache?
If you are going through all the work to look these up, would it make sense to cache the relationship instead